rust-crypto-0.2.31 does not build on 32-bit Xubuntu 14.04, TARGET = Some("i686-unknown-linux-gnu")

When I try to compile rust-crypto-0.2.31 on a target = Some("i686-unknown-linux-gnu") within a cargo build of some project (cargo-version: cargo 0.3.0-nightly (c8c1624 2015-06-26) (built 2015-06-26), rustc-version: rustc 1.3.0-dev (773052a60 2015-06-27)), I get the following errors:

Process didn't exit successfully: `<my_project_home_path>/target/debug/build/rust-crypto-017070612ce33417/build-script-build` (exit code: 101)
--- stdout
TARGET = Some("i686-unknown-linux-gnu")
TARGET = Some("i686-unknown-linux-gnu")
CARGO_MANIFEST_DIR = Some("<my_home_path>/.cargo/registry/src/github.com-121aea75f9ef2ce2/rust-crypto-0.2.31")
OUT_DIR = Some("<my_project_home_path>/target/debug/build/rust-crypto-017070612ce33417/out")
OPT_LEVEL = Some("0")
PROFILE = Some("debug")
debug 0
TARGET = Some("i686-unknown-linux-gnu")
HOST = Some("i686-unknown-linux-gnu")
CC_i686-unknown-linux-gnu = None
CC_i686_unknown_linux_gnu = None
HOST_CC = None
CC = None
TARGET = Some("i686-unknown-linux-gnu")
HOST = Some("i686-unknown-linux-gnu")
CFLAGS_i686-unknown-linux-gnu = None
CFLAGS_i686_unknown_linux_gnu = None
HOST_CFLAGS = None
CFLAGS = None
TARGET = Some("i686-unknown-linux-gnu")
HOST = Some("i686-unknown-linux-gnu")
CC_i686-unknown-linux-gnu = None
CC_i686_unknown_linux_gnu = None
HOST_CC = None
CC = None
running: "cc" "-O0" "-c" "-ffunction-sections" "-fdata-sections" "-m32" "<my_home_path>/.cargo/registry/src/github.com-121aea75f9ef2ce2/rust-crypto-0.2.31/src/util_helpers.c" "-o" "<my_project_home_path>/target/debug/build/rust-crypto-017070612ce33417/out/src/util_helpers.o"
OPT_LEVEL = Some("0")
PROFILE = Some("debug")
debug 0
TARGET = Some("i686-unknown-linux-gnu")
HOST = Some("i686-unknown-linux-gnu")
CC_i686-unknown-linux-gnu = None
CC_i686_unknown_linux_gnu = None
HOST_CC = None
CC = None
TARGET = Some("i686-unknown-linux-gnu")
HOST = Some("i686-unknown-linux-gnu")
CFLAGS_i686-unknown-linux-gnu = None
CFLAGS_i686_unknown_linux_gnu = None
HOST_CFLAGS = None
CFLAGS = None
TARGET = Some("i686-unknown-linux-gnu")
HOST = Some("i686-unknown-linux-gnu")
CC_i686-unknown-linux-gnu = None
CC_i686_unknown_linux_gnu = None
HOST_CC = None
CC = None
running: "cc" "-O0" "-c" "-ffunction-sections" "-fdata-sections" "-m32" "<my_home_path>/.cargo/registry/src/github.com-121aea75f9ef2ce2/rust-crypto-0.2.31/src/aesni_helpers.c" "-o" "<my_project_home_path>/target/debug/build/rust-crypto-017070612ce33417/out/src/aesni_helpers.o"


command did not execute successfully, got: exit code: 1



--- stderr
<my_home_path>/.cargo/registry/src/github.com-121aea75f9ef2ce2/rust-crypto-0.2.31/src/aesni_helpers.c: In function ‘rust_crypto_aesni_aesimc’:
<my_home_path>/.cargo/registry/src/github.com-121aea75f9ef2ce2/rust-crypto-0.2.31/src/aesni_helpers.c:12:5: error: unknown register name ‘xmm1’ in ‘asm’
     asm(
     ^
<my_home_path>/.cargo/registry/src/github.com-121aea75f9ef2ce2/rust-crypto-0.2.31/src/aesni_helpers.c: In function ‘rust_crypto_aesni_setup_working_key_128’:
<my_home_path>/.cargo/registry/src/github.com-121aea75f9ef2ce2/rust-crypto-0.2.31/src/aesni_helpers.c:27:5: error: unknown register name ‘xmm3’ in ‘asm’
     asm(
     ^
<my_home_path>/.cargo/registry/src/github.com-121aea75f9ef2ce2/rust-crypto-0.2.31/src/aesni_helpers.c:27:5: error: unknown register name ‘xmm2’ in ‘asm’
<my_home_path>/.cargo/registry/src/github.com-121aea75f9ef2ce2/rust-crypto-0.2.31/src/aesni_helpers.c:27:5: error: unknown register name ‘xmm1’ in ‘asm’
<my_home_path>/.cargo/registry/src/github.com-121aea75f9ef2ce2/rust-crypto-0.2.31/src/aesni_helpers.c: In function ‘rust_crypto_aesni_setup_working_key_192’:
<my_home_path>/.cargo/registry/src/github.com-121aea75f9ef2ce2/rust-crypto-0.2.31/src/aesni_helpers.c:80:5: error: unknown register name ‘xmm3’ in ‘asm’
     asm(
     ^
<my_home_path>/.cargo/registry/src/github.com-121aea75f9ef2ce2/rust-crypto-0.2.31/src/aesni_helpers.c:80:5: error: unknown register name ‘xmm2’ in ‘asm’
<my_home_path>/.cargo/registry/src/github.com-121aea75f9ef2ce2/rust-crypto-0.2.31/src/aesni_helpers.c:80:5: error: unknown register name ‘xmm1’ in ‘asm’
<my_home_path>/.cargo/registry/src/github.com-121aea75f9ef2ce2/rust-crypto-0.2.31/src/aesni_helpers.c: In function ‘rust_crypto_aesni_setup_working_key_256’:
<my_home_path>/.cargo/registry/src/github.com-121aea75f9ef2ce2/rust-crypto-0.2.31/src/aesni_helpers.c:168:5: error: unknown register name ‘xmm3’ in ‘asm’
     asm(
     ^
<my_home_path>/.cargo/registry/src/github.com-121aea75f9ef2ce2/rust-crypto-0.2.31/src/aesni_helpers.c:168:5: error: unknown register name ‘xmm2’ in ‘asm’
<my_home_path>/.cargo/registry/src/github.com-121aea75f9ef2ce2/rust-crypto-0.2.31/src/aesni_helpers.c:168:5: error: unknown register name ‘xmm1’ in ‘asm’
<my_home_path>/.cargo/registry/src/github.com-121aea75f9ef2ce2/rust-crypto-0.2.31/src/aesni_helpers.c: In function ‘rust_crypto_aesni_encrypt_block’:
<my_home_path>/.cargo/registry/src/github.com-121aea75f9ef2ce2/rust-crypto-0.2.31/src/aesni_helpers.c:266:5: error: unknown register name ‘xmm1’ in ‘asm’
     asm(
     ^
<my_home_path>/.cargo/registry/src/github.com-121aea75f9ef2ce2/rust-crypto-0.2.31/src/aesni_helpers.c:266:5: error: unknown register name ‘xmm0’ in ‘asm’
<my_home_path>/.cargo/registry/src/github.com-121aea75f9ef2ce2/rust-crypto-0.2.31/src/aesni_helpers.c: In function ‘rust_crypto_aesni_decrypt_block’:
<my_home_path>/.cargo/registry/src/github.com-121aea75f9ef2ce2/rust-crypto-0.2.31/src/aesni_helpers.c:303:5: error: unknown register name ‘xmm1’ in ‘asm’
     asm(
     ^
<my_home_path>/.cargo/registry/src/github.com-121aea75f9ef2ce2/rust-crypto-0.2.31/src/aesni_helpers.c:303:5: error: unknown register name ‘xmm0’ in ‘asm’
thread '<main>' panicked at 'explicit panic', <my_home_path>/.cargo/registry/src/github.com-121aea75f9ef2ce2/gcc-0.3.8/src/lib.rs:497

--- a/src/aesni_helpers.c
+++ b/src/aesni_helpers.c
@@ -5,10 +5,12 @@
 // except according to those terms.
 
 #include <stdint.h>
+#include <stdlib.h>
 
 #if defined(__i386__) || defined(__x86_64__)
 
 void rust_crypto_aesni_aesimc(uint8_t* round_keys) {
+    #ifdef __SSE__
     asm volatile(
         " \
             movdqu (%0), %%xmm1; \
@@ -19,11 +21,15 @@ void rust_crypto_aesni_aesimc(uint8_t* round_keys) {
     : "r" (round_keys) // inputs
     : "xmm1", "memory" // clobbers
     );
+    #else
+    exit(1);
+    #endif
 }
 
 void rust_crypto_aesni_setup_working_key_128(
         uint8_t* key,
         uint8_t* round_key) {
+    #ifdef __SSE__
     asm volatile(
         " \
             movdqu (%1), %%xmm1; \
@@ -72,11 +78,15 @@ void rust_crypto_aesni_setup_working_key_128(
     : "r" (key)
     : "xmm1", "xmm2", "xmm3", "memory"
     );
+    #else
+    exit(1);
+    #endif
 }
 
 void rust_crypto_aesni_setup_working_key_192(
         uint8_t* key,
         uint8_t* round_key) {
+    #ifdef __SSE__
     asm volatile(
         " \
             movdqu (%1), %%xmm1; \
@@ -160,11 +170,15 @@ void rust_crypto_aesni_setup_working_key_192(
     : "r" (key)
     : "xmm1", "xmm2", "xmm3", "memory"
     );
+    #else
+    exit(1);
+    #endif
 }
 
 void rust_crypto_aesni_setup_working_key_256(
         uint8_t* key,
         uint8_t* round_key) {
+    #ifdef __SSE__
     asm volatile(
         " \
             movdqu (%1), %%xmm1; \
@@ -256,6 +270,9 @@ void rust_crypto_aesni_setup_working_key_256(
     : "r" (key)
     : "xmm1", "xmm2", "xmm3", "memory"
     );
+    #else
+    exit(1);
+    #endif
 }
 
 void rust_crypto_aesni_encrypt_block(
@@ -263,6 +280,7 @@ void rust_crypto_aesni_encrypt_block(
             uint8_t* input,
             uint8_t* round_keys,
             uint8_t* output) {
+    #ifdef __SSE__
     asm volatile(
     " \
         /* Copy the data to encrypt to xmm1 */ \
@@ -293,6 +311,9 @@ void rust_crypto_aesni_encrypt_block(
     : "r" (input), "r" (output) // inputs
     : "xmm0", "xmm1", "memory", "cc" // clobbers
     );
+    #else
+    exit(1);
+    #endif
 }
 
 void rust_crypto_aesni_decrypt_block(
@@ -300,6 +321,7 @@ void rust_crypto_aesni_decrypt_block(
             uint8_t* input,
             uint8_t* round_keys,
             uint8_t* output) {
+    #ifdef __SSE__
     asm volatile(
         " \
             /* Copy the data to decrypt to xmm1 */ \
@@ -330,6 +352,9 @@ void rust_crypto_aesni_decrypt_block(
     : "r" (input), "r" (output) // inputs
     : "xmm0", "xmm1", "memory", "cc" // clobbers
     );
+    #else
+    exit(1);
+    #endif
 }
 
 #endif

--- a/src/util_helpers.c
+++ b/src/util_helpers.c
@@ -10,6 +10,7 @@
 
 #if defined(__i386__)
 uint32_t rust_crypto_util_supports_aesni() {
+    #ifdef __SSE__
     uint32_t flags;
     asm(
         "pushl %%ebx; \
@@ -20,11 +21,15 @@ uint32_t rust_crypto_util_supports_aesni() {
         : "eax", "edx" // clobbers
     );
     return flags & 0x02000000;
+    #else
+    return 0;
+    #endif
 }
 #endif
 
 #if defined(__x86_64__)
 uint32_t rust_crypto_util_supports_aesni() {
+    #ifdef __SSE__
     uint32_t flags;
     asm(
         "mov $1, %%eax; cpuid;"
@@ -33,6 +38,9 @@ uint32_t rust_crypto_util_supports_aesni() {
         : "eax", "ebx", "edx" // clobbers
     );
     return flags & 0x02000000;
+    #else
+    return 0;
+    #endif
 }
 #endif
 

